extensions/bazel/driver/src/main/scala/org/virtuslab/ideprobe/bazel/BazelProbeDriver.scala

```diff
@@ -1,18 +1,138 @@
+import org.virtuslab.ideprobe.robot.RobotSyntax._
+import scala.concurrent.duration._
 
 object BazelProbeDriver {
   val pluginId = "org.virtuslab.ideprobe.bazel"
 
-  def apply(driver: ProbeDriver): BazelProbeDriver = driver.as(pluginId, new BazelProbeDriver(_))
+  def apply(driver: ProbeDriver): BazelProbeDriver = {
+    driver.as(pluginId, new BazelProbeDriver(_))
+  }
 }
 
-
 class BazelProbeDriver(val driver: ProbeDriver) {
+  private val robotDriver = RobotProbeDriver(driver)
+
   def setupBazelExec(path: Path): Unit = {
     driver.send(BazelEndpoints.SetupBazelExecutable, path)
   }
+
+  def buildBazelProject(): BazelBuildResult = {
+    val robot = robotDriver.robot
+
+    def findBuildResult(): Option[BazelBuildResult] = {
+      val success = "Build completed successfully"
+      val failure = "Build did NOT complete successfully"
+      val messagesToFind = Set(success, failure)
+      val consoles = robot.findAll(query.className("EditorComponentImpl"))
+      val console = consoles.find { console =>
+        messagesToFind.exists { message =>
+          val text = console.fullText
+          text.contains(message)
+        }
+      }
+      console.map { console =>
+        val text = console.fullText
+        BazelBuildResult(isSuccessful = text.contains(success), output = text)
+      }
+    }
+
+    if (robot.findAll(query.className("BaseLabel", "text" -> "Bazel Console")).isEmpty) {
+      robot.find(query.className("StripeButton", "text" -> "Bazel Console")).doClick()
+    }
+
+    driver.invokeActionAsync("MakeBlazeProject")
+
+    var result = Option.empty[BazelBuildResult]
+    val buildWait = WaitLogic.basic(checkFrequency = 1.second) {
+      findBuildResult() match {
+        case buildResult @ Some(_) =>
+          result = buildResult
+          WaitDecision.Done
+        case None => WaitDecision.KeepWaiting("Waiting for bazel build to complete")
+      }
+    }
+
+    driver.await(buildWait)
+
+    result.getOrElse(error("Failed to find build result"))
+  }
+
+  def importProject(
+      importSpec: BazelImportSpec,
+      workspace: Path,
+      waitLogic: WaitLogic
+  ): ProjectRef = {
+    val robot = robotDriver.robot
+    val projectView = prepareBazelProjectViewFile(importSpec, workspace)
+
+    // delete previous project
+    workspace.resolve(".ijwb").delete()
+
+    // start project import wizard
+    driver.invokeActionAsync("Blaze.ImportProject2")
+
+    // set workspace/repository root path
+    robot
+      .find(query.className("TextFieldWithHistory"))
+      .setText(workspace.toRealPath().toString)
+    robot.clickButton("Next")
+
+    // set project view configuration file
+    robot.clickRadioButton("Import project view file")
+    robot.find(query.div("name" -> "projectview-file-path-field")).setText(projectView.toString)
+    robot.clickButton("Next")
+
+    // complete import wizard
+    robot.clickButton("Finish")
+
+    driver.await(waitLogic)
```

Comment:
Should this be a parameter? Should we leave decision about wait logic to the user? Personally, if I was a user,  I would like the implementation to make the right choice

Reply:
When we open project we need some sort of waiting, this is for sure. Default wait logic will wait for indexing and python skeletons update to complete. In benchmarks for example we just wait for the sync to complete by looking at build panel. Also specific parameters of waiting, like basicCheckFrequency in default version are quite long (5 seconds), which seemed to help us with flakiness, but in git-machete they care more about fast test execution and prefer rather tighter check loop and as they import a small project it doesn't matter that much.